File: Maximum Depth of Binary Tree.java
Given the root of a binary tree, return its maximum depth.

A binary tree's maximum depth is the number of nodes along the longest path from the root node down to the farthest leaf node.


class Solution 
{
    public int maxDepth(TreeNode root) 
    {
        // Base case: when it hits null, it means it has no depth.
        if(root == null) return 0;                                   
        
        // The max depth will be either from the left side or right side.
        // We'll take the side that has deeper depth then add 1 since root itself has depth of 1. 
        return 1 + Math.max(maxDepth(root.left), maxDepth(root.right));
    }
}


https://leetcode.com/problems/maximum-depth-of-binary-tree/
Like most of the tree problems, recursion is probably always the approach you want to try.

File: Maximum Depth of Binary Tree.py
https://leetcode.com/problems/maximum-depth-of-binary-tree/
Given the root of a binary tree, return its maximum depth.
A binary tree's maximum depth is the number of nodes along the longest path from the root node down to the farthest leaf node.

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        left_depth = 1 + self.maxDepth(root.left)
        right_depth = 1 + self.maxDepth(root.right)
        
        return max(left_depth, right_depth)

    
 class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right))
